csharp/Adapter/Microsoft.Spark.CSharp/Network/RioNative.cs

```diff
@@ -128,25 +129,34 @@ private void Init()
             var maxThreads = 1;
             if (useThreadPool)
             {
-                maxThreads = Environment.ProcessorCount;
+                maxThreads = (Environment.ProcessorCount + 1) >> 1;
```

Comment:
We separate the work threads to process send and receive completion at the below code. So just make the total number of threads still equals to the machine's processor count.